pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/when/impl/BranchConditional.java

```diff
@@ -46,15 +46,40 @@
  * As populated by {@link jenkins.branch.BranchNameContributor}
  */
 public class BranchConditional extends DeclarativeStageConditional<BranchConditional> {
-    private final String compare;
```

Comment:
Removing the `compare` field might cause deserialization of in flight pipelines to fail when upgrading.

Reply:
I was thinking to normalize the same usage as the `tag` one. Otherwise, users might get confused when using `branch compare: "release-*", comparator: "GLOB"` and `tag pattern: "v-*", comparator: "GLOB"`

What do you suggest?

Reply:
@v1v 
You'll need keep the `compare` field and use a `readResolve` method to migrate data from `compare` to `pattern`.  Look at other instances of `readResolve` in this repo.

Reply:
and
```suggestion
    @Deprecated
    private transient String compare;
```